frontend/components/node-item/node-item.ts

```diff
@@ -33,10 +33,9 @@ export class NodeItem {
 
     // Listen for changes to selected node
     this.componentDataStore.dataStream
-      .pluck('selectedNode')
-      .distinctUntilChanged((selectedNode: any) => {
-        return selectedNode ? selectedNode.id : '';
-      })
+      .map(({ selectedNode }: any) => selectedNode)
+      .filter((selectedNode: any) => selectedNode && selectedNode.id)
+      .distinctUntilKeyChanged('id')
```

Comment:
there is both `distinctUntilChanged ` and `distinctUntilKeyChanged`. Maybe I'm on a newer version of `@reactivex/rxjs`. I'll switch back to `distinctUntilChanged`